terraform/modules/secret_manager/variables.tf

```diff
@@ -23,13 +23,15 @@ variable "labels" {
   default     = {}
 }
 
-variable "gh_pk_accessor_members" {
+variable "gh_pk_accessor_members_map" {
   description = <<EOT
-    The service accounts that need roles/secretmanager.secretAccessor role 
-    for accessing keys stored in secret manager. Normally it will be jvs's
-    api and ui service account.
+    The value of each entry in the map will be the service account that need
+    roles/secretmanager.secretAccessor role for accessing keys stored in secret manager.
+     Normally it will be jvs's api and ui service account.
+    This is a map so when terraform won't complain when values are derived from resource
+    attributes that cannot be determined until apply.
   EOT
-  type = list(string)
+  type = map(string)
```

Comment:
why we need map, can we use https://developer.hashicorp.com/terraform/language/functions/list for the field?

Reply:
Because if we want to get the service account directly from the output of jvs module, like the example code shown in the description link. The service account is not know until `terraform apply`, and terraform will send out error saying the behavior is not deterministic, to make it more deterministic, map is preferred.